### ISSUE TITLE
Contrast Sensitivity and in-game menus

### ISSUE THREAD

**Issue by mryellow** [Anuken/Mindustry-Suggestions#1705]
This is honestly probably heading towards being an Accessibility Bug.

The settings menus and buttons within the game are hard white text on hard black backgrounds.

For myself a few moments of looking at this makes by brain paint high contrast stuff over my vision for the rest of the day or until I spend 30 minutes with eyes closed.

**Comment by DeltaNedas** [comment]
1. where did template go
2. why cant you change your monitor settings
3. how many games have contrast settings

**Comment by mryellow** [comment]
> where did template go

Not sure this fit the template. I was blinded while typing and heading to avoid screen for awhile so just quickly left this in the closet category I could find. I'm sure we can communicate without a template.

> why cant you change your monitor settings

This is about the colour scheme choices. White text on Black background. You'll notice there are few interfaces left in the world which choose such strong contrast.

> how many games have contrast settings

None? That's not the solution, it's just colour choices on buttons and settings screens.

When going for a dark theme the common thing to do is grey it off a little bit, colours are muted rather than stark torches of high contrast light.

**Comment by mryellow** [comment]
This type of stuff, where the high contrast triggers adverse reaction in the viewer. It's why you don't see high-contrast UI like your settings page very often.

> Some migraine sufferers report certain visual patterns can reliably trigger a migraine attack, such as high contrast striped patterns or flickering lights.

https://pubmed.ncbi.nlm.nih.gov/11304022/

There are likely a multitude of conditions which impact people in this way although it surely is a small proportion of the population and an even smaller proportion which would take the time to post about it on your issue tracker.

**Comment by itcannotbe** [comment]
Game triggering adverse reactions?

Make a warning in the splash screen and not play the game. Done.

**Comment by mryellow** [comment]
>  Done.

Seriously? 

I'm talking about a simple aspect of good design. The sort of design you'll see in any dark theme. Contrast being within tolerable ranges.

That's too much to ask and I'm being somehow unreasonable taking up your valuable time?

**Comment by Humanoid-X** [comment]
It's your own problem. Nobody that play Mindustry except you get blind.

My idea is to put a warning when you first open the  game
"WARNING : MAY TRIGGER MIGRAINE OR EPILEPSY EPISODE"

**Comment by mryellow** [comment]
> except you 

No one else has come to the issue tracker and explained the problem.

No one else is likely to.

That doesn't make it stop being an issue with the design. People don't make white on black designs like that anymore because we know better. It might be retrocool, but it's bad accessibility.

It's fine, I can just a get a refund and move on. I really don't care.

The real solution to this real problem though, is to simply take more care, time and effort with design choices.

**Comment by mryellow** [comment]
I have to ask, are the people commenting here involved with the project in any way or just random passers-by hanging shit on design accessibility feedback?

**Comment by mryellow** [comment]
p.s. I hadn't even considered a refund up until the dismissive response here to valid feedback on design accessibility.

**Comment by mryellow** [comment]
I see... It's just random idiots replying, not anyone working on the game. I'll just unsubscribe and leave this open.

**Comment by itcannotbe** [comment]
Ah yes. Contributors are not considered as someone "working on the game.

**Comment by Kyllingene** [comment]
> Ah yes. Contributors are not considered as someone "working on the game.

What contributor here? None.

Also, here is an (oversimplified) way to fix this *without* alienating tons of people:
 - 1: Add an option (or two) in the menu that sets a variable controlling text color. Optionally add another for menu background color.
 - 2: Change things like ```drawText(Colors.red, "Foo")``` to ```drawText(Vars.menuTextColor, "Foo")```. Changing constants to variables is *not* hard, and worth it if it allows people like @mryellow to continue playing.

**Comment by sample-text-here** [comment]
> White text on Black background. You'll notice there are few interfaces left in the world which choose such strong contrast.

1. that's the exact same contrast as black text on a white background
2. there are a lot of programs and websites that do that. it's called "dark mode" and is very popular<sup>[1](#note)</sup>
3. though lower contrast could reduce eye strain, it would be harder to read

<a name="note">1</a>: includes
- discord (default)
- github
- ides
- text editors
- ttys and terminals
- web browsers
- news sites
- social media sites
- operating systems
- image editors

**Comment by mryellow** [comment]
> that's the exact same contrast as black text on a white background

Yeah, it's something to do with the light source being the thing you're focusing on or something. Don't know enough about the technicalities of how the brain deals with it.

> there are a lot of programs and websites that do that. it's called "dark mode" and is very popular1

You'll note every last one of those "dark mode" implementations are some degree of grey on grey rather than hard white on hard black.

> though lower contrast could reduce eye strain, it would be harder to read

There are standards for contrast minimums in web design which may be handy as a reference here.

https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html

**Comment by mryellow** [comment]
> active members of the community. Fairly highly active.

By the way, it was just I wasn't tuned into the fact this is a forum rather than an issue tracker. Has the usual fanboi forum warriors types who feel compelled to voice an opinion on every topic. I should have simply ignored them and I'm sorry biting.

**Comment by mryellow** [comment]
Has it been fixed?

Suggestions never get stale. They remain valid until at the least properly reviewed.

**Comment by itcannotbe** [comment]
Bots are indiscriminate, sadly.
